pkg/pillar/cmd/upgradeconverter/upgradeconverter.go

```diff
@@ -126,3 +135,25 @@ type ConversionHandler struct {
 	description string
 	handlerFunc HandlerFunc
 }
+
+//RunPostVaultHandlers invokes postVaultconversionHandlers
+//and notifies the caller through the provided ucChan channel
+//the channel is useful for calling from other agent modules
+//without missing watchdog, by spawning this as a task, and
+//select()ing for its completion
+func RunPostVaultHandlers(ps *pubsub.PubSub,
+	loggerArg *logrus.Logger,
+	logArg *base.LogObject,
+	debugOverride bool, ucChan chan<- int) {
+	logger = loggerArg
+	log = logArg
+	ctx := &ucContext{agentName: "upgradeconverter",
+		persistDir:       types.PersistDir,
+		persistConfigDir: types.PersistConfigDir,
+		varTmpDir:        "/var/tmp",
+		ps:               ps,
+	}
+	runHandlers(ctx, postVaultconversionHandlers)
+	log.Notice("RunPostVaultHandlers completed, notifying caller")
+	ucChan <- 1
```

Comment:
Go standard is to use a chan struct{} for notifications, and not a chan of some other type.

Reply:
Updated the changes accordingly, please check